src/cholesky.jl

```diff
@@ -0,0 +1,65 @@
+import Base: getproperty
+import LinearAlgebra: cholesky, Cholesky, char_uplo, UpperTriangular, LowerTriangular, \,
+    istril, istriu
+
+const KroneckerCholesky{T} = Cholesky{T, <:AbstractKroneckerProduct{T}} where {T}
+
+function cholesky(A::AbstractKroneckerProduct; check=true)
+    P, Q = getmatrices(A)
+    chol_P, chol_Q = cholesky(P; check=true), cholesky(Q; check=true)
+    return Cholesky(chol_P.factors ⊗ chol_Q.factors, 'U', 0)
+end
+
+function Cholesky(factors::KroneckerProduct{T}, uplo::AbstractChar, info::Integer) where {T}
+    return Cholesky{T, typeof(factors)}(factors, uplo, info)
+end
+
+function UpperTriangular(C::AbstractKroneckerProduct)
+    A, B = getmatrices(C)
+    return UpperTriangular(A) ⊗ UpperTriangular(B)
+end
+
+function LowerTriangular(C::AbstractKroneckerProduct)
+    A, B = getmatrices(C)
+    return LowerTriangular(A) ⊗ LowerTriangular(B)
+end
+
+function istril(C::AbstractKroneckerProduct)
+    A, B = getmatrices(C)
+    return istril(A) && istril(B)
+end
+
+function istriu(C::AbstractKroneckerProduct)
+    A, B = getmatrices(C)
+    return istriu(A) && istriu(B)
+end
+
+function getproperty(C::KroneckerCholesky, d::Symbol)
+    Cfactors = getfield(C, :factors)
+    Cuplo    = getfield(C, :uplo)
+    info     = getfield(C, :info)
+
+    Cuplo != 'U' && throw(NotImplementedError(""))
+
+    if d == :U
+        return UpperTriangular(Cuplo === char_uplo(d) ? Cfactors : copy(Cfactors'))
+    elseif d == :L
+        return LowerTriangular(Cuplo === char_uplo(d) ? Cfactors : copy(Cfactors'))
+    elseif d == :UL
+        return (Cuplo === 'U' ? UpperTriangular(Cfactors) : LowerTriangular(Cfactors))
+    else
+        return getfield(C, d)
+    end
+end
+
+function logdet(C::KroneckerCholesky)
+    A, B = getmatrices(C.factors)
+    logdet_A = logdet(Cholesky(A, C.uplo, 0))
+    logdet_B = logdet(Cholesky(B, C.uplo, 0))
+    return size(B, 1) * logdet_A + size(A, 1) * logdet_B
+end
+
+function \(C::KroneckerCholesky, x::AbstractVecOrMat)
```

Comment:
I am not sure this is correct. If I understand it correctly, the main performance of Cholesky decomposition originates from the fact that it is easy to solve a triangular system. The vec trick kind of destroys this structure...

Reply:
Hmm. I admit that I hadn't really thought very hard about this when writing it, but I'm pretty sure that you can use the vec trick to compute backsolves. Note that
```julia
vec(inv(L2) X inv(L1)) = (inv(L1) ⊗ inv(L2)) vec(X) = inv(L1 ⊗ L2) vec(X)
```
so we should be able to implement backsolving with kronecker matrices efficiently provided that the individual backsolves are efficient.